_sections/30-bindings.md

```diff
@@ -112,8 +112,8 @@ A subset of [XPath 1.0 functions](http://www.w3.org/TR/xpath/#corelib), some fun
 <a id="fn:boolean-from-string" href="#fn:boolean-from-string">`boolean-from-string(string arg)`</a> | Returns true if arg is "true" or "1", otherwise returns false.
 <a id="fn:not" href="#fn:not">`not(boolean arg)`</a> | As in [XPath 1.0](http://www.w3.org/TR/xpath/#function-not).
 <a id="fn:number" href="#fn:number">`number(* arg)`</a> | As in [XPath 1.0](http://www.w3.org/TR/xpath/#function-number).
-<a id="fn:decimal-date-time" href="#fn:decimal-date-time">`decimal-date-time(dateTime value)`</a> | Converts dateTime value to a number.
-<a id="fn:decimal-time" href="#fn:decimal-time">`decimal-time(time value)`</a> | Converts time value to a number.
+<a id="fn:decimal-date-time" href="#fn:decimal-date-time">`decimal-date-time(dateTime value)`</a> | Converts dateTime value to a number in the format used by Excel. The number before the decimal marker is the number of days since January 1, 1970 UTC. The number after the decimal marker represents the time as a fractional day.
+<a id="fn:decimal-time" href="#fn:decimal-time">`decimal-time(time value)`</a> | Converts time value to a number representing a fractional day. For example, noon is 0.5 and 6pm is 0.75.
```

Comment:
Where this becomes unclear is how to convert time zones. My guess is it should be converted to UTC and use negative values if you move backwards past midnight UTC, and use a value larger than 1 if you move forward past midnight UTC.

Reply:
I feel strongly that there should be no negatives. In clock time, 00:00 and 24:00 are the same thing. Similarly, in this scheme 0 and 1 are the same.

The behavior that has existed since the introduction of this function in JavaRosa is to ignore timezone. I think in this case it does make sense because the function exists to allow calculations on raw times. I think the primary use case is subtracting one time from another to get an interval. Things are a lot easier to reason about without having to consider timezone conversions.

Reply:
I would have agreed with you until about a month ago when I learned that [ODK Collect's time format included the timezone](https://github.com/enketo/enketo-core/issues/463) and that Enketo was embarrassingly not following the spec (rectified now). Stripping timezone info away in decimal-time() feels wrong to me. Isn't the addition of timezones in JavaRosa's time format recent also?

@alxndrsn, @tiritea, any opinions?

Reply:
Ooh, actually, I was wrong about the existing behavior. It's not stripped. Time is converted to the current timezone, as in Excel (https://en.wikipedia.org/wiki/Decimal_time#Fractional_days -- "Microsoft Excel uses the local time zone of the computer").

I'm currently in UTC-8. If I have a timestamp of 16:00:00.000-8 and call `decimal-time` on it, I get 0.6666666666678793. If I have a timestamp of 16:00:00.000-12, I get 0.8333333333321207 (16:00-12 is 08:00-8).

It does make sense to normalize to one timezone now that I think about it. Definitely interested in what others think. Timezones give me headaches.